app/helpers/info_request_helper.rb

```diff
@@ -335,4 +335,11 @@ def attachment_params(attachment, options = {})
   def public_token?
     defined?(public_token) && public_token.present?
   end
+
+  def fetch_foi_attachments(info_request)
```

Comment:
At the moment this won't be very performant, and is kind of in the wrong place. Helpers are more for presentation logic than querying and computation.

I think we could replace most of this with a [`has_many :through` association](https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association). That way Rails takes care of all the SQL joins and – I haven't tested this so I might have missed a gotcha – should allow us to do `@info_request.foi_attachments`.

We'd then want to filter out the main body part. That would be better as a [`scope`](https://guides.rubyonrails.org/active_record_querying.html#scopes) on `FoiAttachment`:

```rb
scope :substantive, -> { reject(&:main_body_part?) }
```

Then we could chain that on to our previous improvement and end up with `@info_request.foi_attachments.substantive`. We'd want to add a test for this as there's a little more logic involved than plugging together the Rails components.

We can then just call that directly when rendering the attachments:

```erb
<%= render partial: 'admin_incoming_message/foi_attachments',
           locals: { foi_attachments: @info_request.foi_attachments.substantive } %>
```